packages/@aws-cdk/custom-resources/lib/aws-custom-resource/runtime/index.ts

```diff
@@ -105,13 +105,33 @@ export async function handler(event: AWSLambda.CloudFormationCustomResourceEvent
 
     let flatData: { [key: string]: string } = {};
     let data: { [key: string]: string } = {};
-    const call: AwsSdkCall | undefined = event.ResourceProperties[event.RequestType];
+    const call: EncodedAwsSdkCall | undefined = event.ResourceProperties[event.RequestType];
 
     if (call) {
-      const awsService = new (AWS as any)[call.service]({
-        apiVersion: call.apiVersion,
-        region: call.region,
-      });
+
+      let awsService: any;
+      if(call.assumedRole){
```

Comment:
Please adhere to the surrounding style:

```suggestion
      if (call.assumedRole) {
```